Code/Framework/AzCore/AzCore/RTTI/BehaviorContext.h

```diff
@@ -313,7 +315,11 @@ namespace AZ
         AZStd::string m_deprecatedName;     ///<this is the deprecated name of this method
         const char* m_debugDescription;
         bool m_isConst = false; ///< Is member function const (false if not a member function)
+        int m_startArgumentIndex = 1; // +1 for result type
         AttributeArray m_attributes;    ///< Attributes for the method
+        AZStd::vector<BehaviorParameterMetadata> m_metadataParameters; ///< Stores the per parameter metadata which is used to add names, tooltips, trait, default values, etc...
```

Comment:
This changes from a fixed size to a dynamically sized container. While that will be reduce the application size, but wouldn't that also negatively impact performance?

Reply:
These parameters are only used by the editor, there is no performance concern for runtime. That said, I do not believe the functionality is stripped in a runtime build so there is a cost of populating this container during reflection, which would manifest during startup, it's unlikely to be a significant portion of the load time, though we'd need to profile to be certain.

Reply:
The resize also happens once during the creation of the "Class" object